lib/takeaway.rb

```diff
@@ -0,0 +1,73 @@
+require_relative 'order'
+require_relative 'sms'
+
+module Menu
+  def print_menu(dishes)
+    puts dishes.map { |dish| format(dish) }.join("\n\n")
+  end
+end
+# following SRP
+
+class Takeaway
+  attr_reader :current_order
+  include Menu
+  
+  @@sms_client = SMS.new
+
+  def initialize(dishes, order_class: Order)
+    dishes = [dishes] unless dishes.is_a? Array # can pass in 1 or more dishes
+    @dishes = dishes
```

Comment:
Could this be more descriptively named? Are the dishes chosen by the user or are they the dishes available?